src/transport_performance/gtfs/validation.py

```diff
@@ -601,3 +655,803 @@ def get_route_modes(self):
+                impacted_rows, scheme, padding="10px", escape=False
+            )
+
+            table_html = table_html + "</body>"
+
+            # save the output
+            save_name = f"{'_'.join(message.split(' '))}_{table}"
+            with open(f"{output_path}/gtfs_report/{save_name}.html", "w") as f:
+                f.write(table_html)
+
+        return None
+
+    def html_report(
+        self,
+        report_dir: Union[str, pathlib.Path] = "outputs",
+        overwrite: bool = False,
+        summary_type: str = "mean",
+        extended_validation: bool = True,
+    ) -> None:
+        """Generate a HTML report describing the GTFS data.
+
+        Parameters
+        ----------
+        report_dir : Union[str, pathlib.Path], optional
+            The directory to save the report to,
+            by default "outputs"
+        overwrite : bool, optional
+            Whether or not to overwrite the existing report
+            if it already exists in the report_dir,
+            by default False
+        summary_type : str, optional
+            The type of summary to show on the
+            summaries on the gtfs report.,
+            by default "mean"
+        extended_validation : bool, optional
+            Whether or not to create extended reports
+            for gtfs validation errors/warnings.
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValueError
+            An error raised if the type of summary passed is invalid
+
+        """
+        _type_defence(overwrite, "overwrite", bool)
+        _type_defence(summary_type, "summary_type", str)
+        set_up_report_dir(path=report_dir, overwrite=overwrite)
+        summary_type = summary_type.lower()
+        if summary_type not in ["mean", "min", "max", "median"]:
+            raise ValueError("'summary type' must be mean, median, min or max")
+
+        # store todays date
+        date = datetime.datetime.strftime(datetime.datetime.now(), "%d-%m-%Y")
+
+        # feed evaluation
+        self.clean_feed()
+        validation_dataframe = self.is_valid()
+
+        # create extended reports if requested
+        if extended_validation:
+            self._extended_validation(output_path=report_dir)
+            info_href = (
+                validation_dataframe["message"].apply(
+                    lambda x: "_".join(x.split(" "))
+                )
+                + "_"
+                + validation_dataframe["table"]
+                + ".html"
+            )
+            validation_dataframe["info"] = [
+                f"""<a href="{href}"> Further Info</a>"""
+                if len(rows) > 1
+                else "Unavailable"
+                for href, rows in zip(info_href, validation_dataframe["rows"])
+            ]
+
+        eval_temp = TemplateHTML(
+            path=(
+                "src/transport_performance/gtfs/report/"
+                "html_templates/evaluation_template.html"
+            )
```

Comment:
From a functional purity perspective, we should pass this path in as a value to the method. Danger being that breaking that external dependency doesn’t get handled at instantiation / in defensive checks.

Reply:
On reflection I agree with your position on this - not wishing to parameterise it incase we encourage non-compatible alternative templates.